test/geometry_tests.cpp

```diff
@@ -45,6 +54,39 @@ bool allXOrYCoorSame(int x_or_y, std::vector<std::vector<int>> pixels){
 
 }
 
+bool allXOrYCoorSame(int x_or_y, std::vector<Cell> v){
```

Comment:
You should make a matcher for this https://github.com/google/googletest/blob/master/googlemock/docs/cheat_sheet.md#matchers-matcherlist

Reply:
I don't understand the point of this function

Reply:
Didn't we discuss getting rid of this function?

Reply:
Indubitably